src/Microsoft.PowerShell.Commands.Utility/commands/utility/UnblockFile.cs

```diff
@@ -119,18 +119,9 @@ protected override void ProcessRecord()
                     {
                         AlternateDataStreamUtilities.DeleteFileStream(path, "Zone.Identifier");
                     }
-                    catch (Win32Exception accessException)
+                    catch (Exception accessException)
```

Comment:
I recommend not using Exception here; it's a bad practice.  DeleteFileStream throws Win32Exception for File IO errors and those are what you want to catch.  you can avoid the ArgumentNullException by annotation the cmdlet argument property to require non-null. that will provide a better user error than ArgumentNullException.

Reply:
> you can avoid the ArgumentNullException 

Path and LiteralPath parameters is _mandatory_ in the cmdlet that require non-null.

> I recommend not using Exception here; it's a bad practice

I started with this but revert the change because seems we always _wrap_ system exceptions (like Win32Exception) in PowerShell custom ones to get a customized message and keep system exception as InnerException. 
Why should we do otherwise here?

Reply:
The questions you should ask yourself...

* Will you provide a usable error message for every type of exception that can be throw?
Should every cmdlet implement a policy that rightfully belongs in PowerShell?

This question comes up over and over and, too often, it's punted by using catch (Exception).  
In my opinion, if there are specific exceptions where you can provide value (targeted user message, retry logic, fallback logic, etc.), then, of course, handle the exception.  

If you cannot, then you should allow PowerShell to provide common behavior across cmdlets instead of replicating code and inconsistent logic.

Here's a good discussion about why it's often bad to catch Exception: 
https://blogs.msdn.microsoft.com/dotnet/2009/02/19/why-catchexceptionempty-catch-is-bad/

Reply:
@dantraMSFT Thanks! I didn't know that C# caught all system exceptions.

If we remove the try-catch the error message will be different only in `CategoryInfo` (PermissionDenied -> NotSpecified):
```
// before
writeErrorStream      : True
PSMessageDetails      :
Exception             : System.UnauthorizedAccessException: Access to the path 'C:\tmp\q.txt:Zone.Identifier' is
                        denied.
                           at System.IO.FileSystem.DeleteFile(String fullPath)
                           at System.IO.File.Delete(String path)
                           at
                        System.Management.Automation.Internal.AlternateDataStreamUtilities.DeleteFileStream(String
                        path, String streamName) in C:\Users\sie\Documents\GitHub\iSazonov\PowerShell\src\System.Manage
                        ment.Automation\namespaces\FileSystemProvider.cs:line 8544
                           at Microsoft.PowerShell.Commands.UnblockFileCommand.ProcessRecord() in C:\Users\sie\Document
                        s\GitHub\iSazonov\PowerShell\src\Microsoft.PowerShell.Commands.Utility\commands\utility\Unblock
                        File.cs:line 120
TargetObject          : C:\tmp\q.txt
CategoryInfo          : PermissionDenied: (C:\tmp\q.txt:String) [Unblock-File], UnauthorizedAccessException
FullyQualifiedErrorId : RemoveItemUnauthorizedAccessError,Microsoft.PowerShell.Commands.UnblockFileCommand
ErrorDetails          :
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {0, 1}

-----------------------------
// after removing try-catch
PSMessageDetails      :
Exception             : System.UnauthorizedAccessException: Access to the path 'C:\tmp\q.txt:Zone.Identifier' is
                        denied.
                           at System.IO.FileSystem.DeleteFile(String fullPath)
                           at System.IO.File.Delete(String path)
                           at
                        System.Management.Automation.Internal.AlternateDataStreamUtilities.DeleteFileStream(String
                        path, String streamName) in C:\Users\sie\Documents\GitHub\iSazonov\PowerShell\src\System.Manage
                        ment.Automation\namespaces\FileSystemProvider.cs:line 8544
                           at Microsoft.PowerShell.Commands.UnblockFileCommand.ProcessRecord() in C:\Users\sie\Document
                        s\GitHub\iSazonov\PowerShell\src\Microsoft.PowerShell.Commands.Utility\commands\utility\Unblock
                        File.cs:line 118
                           at System.Management.Automation.Cmdlet.DoProcessRecord() in C:\Users\sie\Documents\GitHub\iS
                        azonov\PowerShell\src\System.Management.Automation\engine\cmdlet.cs:line 172
                           at System.Management.Automation.CommandProcessor.ProcessRecord() in C:\Users\sie\Documents\G
                        itHub\iSazonov\PowerShell\src\System.Management.Automation\engine\CommandProcessor.cs:line 358
TargetObject          :
CategoryInfo          : NotSpecified: (:) [Unblock-File], UnauthorizedAccessException
FullyQualifiedErrorId : System.UnauthorizedAccessException,Microsoft.PowerShell.Commands.UnblockFileCommand
ErrorDetails          :
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {}

```
We could catch only IOException and UnauthorizedAccessException exception.
Although I agree to remove it.